### ISSUE TITLE
Stops responding when editing a large note on Android

### ISSUE THREAD

**Issue by da2x** [laurent22/joplin#2359]
## Environment

Joplin version: 1.0.312
Platform: Android
OS specifcs: 10, Samsung S10e

## Steps To Reproduce

1. Create a new note and *quickly* type a 4000+ character note in one go

The app stops responding and text input no longer appears on screen. If you the problemt and continue typing, eventually the on-screen keyboard will also stop giving visual indications as to what letters you press. The whole system becomes unresponsive. Wait up to 10 seconds and the app will catch up with what you've typed and recover.

After the app has lagged behind input once, the app will be slow to respond to all subsequent text input. You can temporarily work around the issue by quitting the app and editing the same note. Text editing will then be fast for a short while. After some time, the issue reappears and text input becomes low and unresponsive again.

**Comment by virtadpt** [comment]
I see this quite often, also.

**Comment by bedwardly-down** [comment]
@da2x I have a series of questions for you and things for you to try:

1. When was the last time you restarted your phone?
2. When was the last time you went to Settings->System Updates->Download and install to see if there are new updates for your phone from Samsung?
3. When was the last time you went to Settings->Device Care->Memory and ran Clean Now (to stop background apps you're not using that may be hogging valuable resources)?
4. What launcher (Home UI) are you currently using?
5. Have you uninstalled and reinstalled Joplin from the Play Store?
6. How do you type a 4000+ note in one go without your thumbs getting tired? O.o

**Comment by ArjenB96** [comment]
Might be related to #2357.

**Comment by shubhan-shu** [comment]
I am experiencing the same problem. Whenever the keyboard starts to lag after working in Joplin for some time I clear the data of the keyboard I am using(Gboard) from the Settings. This fixes the issue. A workaround for the problem is to use NeutriNote app on Android to write notes in markdown and then share it to Joplin or copy-paste the markdown code.

**Comment by ivywong** [comment]
I believe this is the same issue as #1493; it's an issue with React Native itself (to be potentially fixed in v0.62)

**Comment by da2x** [comment]
It’s still a problem with the most recent version.

**Comment by virtadpt** [comment]
The problem seems to be in an upstream JS library.  There's a thread on the forums about it.

tl;dr - Android is a no-go until that gets fixed.

**Comment by da2x** [comment]
Still an issue.

**Comment by bedwardly-down** [comment]
Reopening due to bug still showing up. Anyone have any thoughts on how to solve this?

**Comment by shreyasbharath** [comment]
Still a major issue, app is unusable if notes are large enough.

**Comment by vlig** [comment]
VERY slow on editing quite large notes on Android.

**Comment by Jonta** [comment]
> The problem seems to be in an upstream JS library. There's a thread on the forums about it.

@virtadpt Link? =)

**Comment by virtadpt** [comment]
@Jonta https://discourse.joplinapp.org/t/editing-a-long-note-is-slow-on-android/4441

**Comment by Jonta** [comment]
Thanks

That one mentions React Native-version, which currently is 0.59 or 0.59.5, found in `packages/app-mobile/package-lock.json`:

https://github.com/laurent22/joplin/blob/33822295ef3295954ad1b5a61ad2d5260336f9de/packages/app-mobile/package-lock.json#L1905
https://github.com/laurent22/joplin/blob/33822295ef3295954ad1b5a61ad2d5260336f9de/packages/app-mobile/package-lock.json#L1923
https://github.com/laurent22/joplin/blob/33822295ef3295954ad1b5a61ad2d5260336f9de/packages/app-mobile/package-lock.json#L9815


And links to this thread: https://github.com/facebook/react-native/issues/20119#issuecomment-576917362

**Comment by hoppushoppard** [comment]
> I'm experiencing same issue. After writing even less than hundred words (into the new note), app starts being unresponsive, than keyboard lags until couple of seconds, when it catches up. Eventually i have to leave editing mode. And enter it again.
> 
> Android 11 (Oxygen OS 11.1.5.5) OnePlus Nord
> 
> Microsoft SwiftKey keyboard.
> 
> Joplin 2.4.3 Database v39 FTS Enabled 1

Solution for me was to opt-in to new editor beta

**Comment by bataliero** [comment]
> I'm experiencing same issue. After writing even less than hundred words (into the new note), app starts being unresponsive, than keyboard lags until couple of seconds, when it catches up. Eventually i have to leave editing mode. And enter it again.
> 
> Android 11 (Oxygen OS 11.1.5.5) OnePlus Nord
> 
> Microsoft SwiftKey keyboard.
> 
> Joplin 2.4.3 Database v39 FTS Enabled 1

I have the same issue. As a temporary solution, I am switching from `Microsoft SwiftKey keyboard.` to `Gboard` while note starts to become laggy. That makes I can finish typing my note : ) but anyway still annoying bug.

Android 11 
OnePlus Nord 2

Joplin 2.8.1
Database v41
FTS enabled: 1

**Comment by shadowbane1000** [comment]
I was experiencing what i believe is this same issue.  I open a note that isn't very large (maybe 35 lines), and as I type things go slower in what appears to be an exponential progression.  

I tried switching from the 2.8.1 release to the new 2.10 release (released today) and the same thing occurred.  Since the editor has been rewritten, I figured it must be something else.  My phone is also a Samsung (S21 FE) like the original reporter.

On a whim, I switched the keyboard from the Samsung keyboard to GBoard, and presto, no slowdown.  Switch back, and it slows down again.  So it is, in my case, some interaction with the keyboard.  

It also seems that I can append some information to the note without a slowdown, but inserting information in to the note takes only a handful of characters to become unusable.  Inserting one character, then appending to the end of the note seems to continue the exponential slowdown. 

To escape from the slowdown I need only exit editing the note and re-enter editing the note.

I hope this helps.
